packages/workspace/src/workspace/workspace.ts

```diff
@@ -290,22 +286,25 @@ export const loadWorkspace = async (config: ConfigSource, credentials: ConfigSou
         currentEnv(), elementsSources.commonSourceName)
     },
     renameEnvironment: async (envName: string, newEnvName: string) => {
-      const envConfig = workspaceConfig.envs.find(e => e.name === envName)
+      const envConfig = envs().find(e => e === envName)
       if (_.isUndefined(envConfig)) {
         throw new UnknownEnvError(envName)
       }
 
-      if (!_.isUndefined(workspaceConfig.envs.find(e => e.name === newEnvName))) {
+      if (!_.isUndefined(envs().find(e => e === newEnvName))) {
         throw new EnvDuplicationError(newEnvName)
       }
 
-      envConfig.name = newEnvName
-      await config.set(WORKSPACE_CONFIG_NAME, workspaceConfigInstance(workspaceConfig))
-
-      if (envName === userConfig.currentEnv) {
-        userConfig.currentEnv = newEnvName
-        await config.set(USER_CONFIG_NAME, workspaceUserConfigInstance(userConfig))
+      currentEnvsConf().map(e => {
+        if (e.name === envName) {
+          e.name = newEnvName
+        }
+        return e
+      })
```

Comment:
should not use `map` for a side effect while ignoring the return value.
I think it was better the way it was before this change
```suggestion
      const envConfig = currentEnvsConf().find(e => e.name === envName)
      // ... do the `undefined` check on this instead of how it is currently written
      envConfig.name = newEnvName
```
but if you want to change it, at least change the `map` to `forEach` and maybe use `filter` instead of `if` in the function body?

Reply:
Fixed